core/common/modules/connect/admin.php

```diff
@@ -30,21 +31,27 @@ public function register_admin_menu( Admin_Menu_Manager $admin_menu ) {
 	public function on_load_page() {
 		if ( isset( $_GET['action'], $_GET['app'] ) ) {
 			$manager = Plugin::$instance->common->get_component( 'connect' );
-			$app_slug = $_GET['app'];
+
+			// phpcs:ignore WordPress.Security.NonceVerification.Recommended
+			$app_slug = Utils::get_super_global_value( $_GET, 'app' );
 			$app = $manager->get_app( $app_slug );
-			$nonce_action = $_GET['app'] . $_GET['action'];
+
+			// phpcs:ignore WordPress.Security.NonceVerification.Recommended
+			$action = Utils::get_super_global_value( $_GET, 'action' );
+
+			$nonce_action = $app_slug . $action;
 
 			if ( ! $app ) {
 				wp_die( 'Unknown app: ' . esc_attr( $app_slug ) );
 			}
 
-			if ( empty( $_GET['nonce'] ) || ! wp_verify_nonce( $_GET['nonce'], $nonce_action ) ) {
+			if ( ! wp_verify_nonce( Utils::get_super_global_value( $_GET, 'nonce' ), $nonce_action ) ) {
```

Comment:
Maybe move it up instead of ignore the nonce verification

Reply:
It had to be there since it uses the `$nonce_action` variable